### ISSUE TITLE
generate_method_stub emits bare keyword for @-escaped call-site method name → SUCCESS + CS1001/CS1519 (ResolveMethodName ValueText / CreateMethodStub)

### ISSUE THREAD

**Issue by JoshuaRamirez** [JoshuaRamirez/RoslynMcpServer#1312]
## Summary

`generate_method_stub` reports **SUCCESS** but emits a **bare reserved keyword** as the method declaration name when the undefined call site uses an `@`-escaped identifier (`@class()` → `private void class()`). `dotnet build` then fails with **CS1519** / **CS1001** (Invalid token 'class' in a member declaration / Identifier expected).

Root cause is two seams:
1. `ResolveMethodName` (~L637–651) returns `invokedName.Identifier.ValueText`, which strips the `@` escape (`@class` → `class`).
2. Local `CreateMethodStub` (~L1237–1239) builds `SyntaxFactory.MethodDeclaration(..., name)` with that bare string — no `@` / verbatim identifier.

Control with a non-keyword call site (`DoWork()`) → **SUCCESS** + clean `dotnet build` (**WRONG=no**).

Distinct from `#1181` (named `@`-escaped **parameter** / `NameColon` → bare param name). Distinct from validation-only `#475` (explicit reserved **methodName** param → wrong error code, not SUCCESS + illegal emit). Distinct from member-name bare-keyword bugs on other ops (`#1303`/`#1304`/`#1305`/`#1297`) — those copy `ISymbol.Name` from existing members; this invents the name from the call-site invocation identifier.

## Master SHA

`6c4a6f04d72625da442c9c6559c8d7de65e0cff6` (post `#1308` introduce_field allFiles; open leftover count 0).

## Repro

```csharp
#nullable enable
namespace TestApp;
public class Widget
{
    public void Run()
    {
        @class();
    }
}
```

1. Baseline: undefined `@class` (expected; stub target).
2. `generate_method_stub` on the `@class(...)` call site → **SUCCESS**.
3. Emitted member: `private void class() { throw new global::System.NotImplementedException(); }` (**no** `@` on the method name).
4. `dotnet build` → **CS1519** / **CS1001** (and follow-on parse errors) — **WRONG=yes**.
5. Control: same with `DoWork()` → SUCCESS + `private void DoWork()` + clean build — **WRONG=no**.

## Actual

| Shape | Op | Result | Rewrite | `dotnet build` |
| --- | --- | --- | --- | --- |
| `@class()` call-site keyword | `generate_method_stub` | SUCCESS | `private void class()` | **CS1519 / CS1001** |
| `DoWork()` call-site control | `generate_method_stub` | SUCCESS | `private void DoWork()` | clean |

Live emit (SUCCESS):

```csharp
private void class()
{
    throw new global::System.NotImplementedException();
}
```

## Expected

- Method names inferred from `@`-escaped call-site identifiers must be emitted as verbatim identifiers (`@class`), matching the call site.
- `generate_method_stub` on `@class()` → SUCCESS + compile-clean stub `private void @class() { ... }`, or a clear validation failure instead of SUCCESS + illegal syntax.
- Keep non-keyword call-site control behavior.

## Root cause

`src/RoslynMcp.Core/Refactoring/Generate/GenerateMethodStubOperation.cs` — `ResolveMethodName` (~L637–651):

```csharp
return invokedName.Identifier.ValueText; // "@class" → "class"
```

Then local `CreateMethodStub` (~L1237–1239):

```csharp
var method = SyntaxFactory.MethodDeclaration(
        SyntaxFactory.ParseTypeName(returnType).WithTrailingTrivia(SyntaxFactory.Space),
        name) // bare "class" → MethodDeclaration identifier without @
```

`SyntaxFactory.MethodDeclaration(TypeSyntax, string)` / `Identifier(string)` does not add `@` for reserved keywords.

## Control WRONG=no

`DoWork()` → SUCCESS + `private void DoWork()` + `dotnet build` exit 0.

## Related siblings (do not duplicate)

| Issue | What it covers |
| --- | --- |
| #1181 | `generate_method_stub` bare keyword for named `@`-escaped **args** (`CreateParameter` / `NameColon`) |
| #475 | `generate_method_stub` reserved **methodName** param → wrong error code (validation-only) |
| #1303 / #1304 / #1305 / #1297 | Other ops bare keyword for `@`-escaped **member names** from `ISymbol.Name` |
| #1182 / #1176 / #1302 | Other ops bare keyword for `@`-escaped **params** |

## Separate follow-ups (not this bug)

- `InferTypeParameters` inventing `T`/`T2` without constraints from call-site type args — inventing unconstrained type params is valid C# (**WRONG=no** for compile).
- `implement_abstract` omitting `ConstraintClauses` — overrides inherit constraints (**WRONG=no** per `#1196` body); skip.
- Explicit `methodName` reserved-keyword validation mapping (`#475`).

## Suggested fix

When emitting the method identifier (and when taking `ValueText` from the call site), if `SyntaxFacts.GetKeywordKind(name) != SyntaxKind.None`, use a verbatim identifier (`SyntaxFactory.Identifier("@" + name)` or shared helper). Prefer the same escape helper used for peer member/param `@` fixes.

## Labels

`bug` only (open leftover count 0; not exclusive-end class).

## Confidence

High — live `GenerateMethodStubOperation` `Success=True` + `dotnet build` CS1519/CS1001 on master `6c4a6f0` (bug `@class()`; control `DoWork()` clean).
